pkg/diagnose/runner.go

```diff
@@ -407,45 +435,88 @@ func runStdOut(w io.Writer, diagCfg diagnosis.Config, run func(diagnosis.Config)
 		color.NoColor = true
 	}
 
-	fmt.Fprintf(w, "=== Starting diagnose ===\n")
+	if !diagCfg.JSONOutput {
+		fmt.Fprintf(w, "=== Starting diagnose ===\n")
+	}
 
 	diagnoses, err := run(diagCfg)
-	if err != nil && !diagCfg.RunLocal {
-		fmt.Fprintln(w, color.YellowString(fmt.Sprintf("Error running diagnose in Agent process: %s", err)))
-		fmt.Fprintln(w, "Running diagnose command locally (may take extra time to run checks locally) ...")
-
-		// attempt to do so locally
+	if err != nil {
+		errMsg := fmt.Sprintf("Error running diagnose in Agent process: %s", err)
+		action := "Running diagnose command locally (may take extra time to run checks locally) ..."
+		if diagCfg.JSONOutput {
+			errJSON, _ := json.MarshalIndent(map[string]string{
+				"error":  errMsg,
+				"action": action,
+			}, "", "  ")
+			fmt.Fprintln(w, string(errJSON))
+		} else if !diagCfg.RunLocal {
+			fmt.Fprintln(w, color.YellowString(errMsg))
+			fmt.Fprintln(w, action)
+		}
+		// Attempt to run diagnose locally
 		diagCfg.RunLocal = true
 		diagnoses, err = run(diagCfg)
 	}
 
 	if err != nil {
```

Comment:
question:

why do we have 2 consecutive if statement blocks with the same condition? can't we simply merge them into one block?

Reply:
This is because a new error can be created at `diagnoses, err = run(diagCfg)`.